emuu-client/src/components/home/Home.js

```diff
@@ -81,10 +77,10 @@ function Home() {
   function sortVideosByTime(videos) {
     for (let i = 0; i < videos.length - 1; i++) {
       for (let j = 0; j < videos.length - 1 - i; j++) {
-        if (videos[i].data().uploadTime < videos[i + 1].data().uploadTime) {
-          let temp = videos[i];
-          videos[i] = videos[i + 1];
-          videos[i + 1] = temp;
+        if (videos[j].uploadTime < videos[j + 1].uploadTime) {
+          let temp = videos[j];
+          videos[j] = videos[j + 1];
+          videos[j + 1] = temp;
```

Comment:
Good catch that having [i] as the index outputted the wrong information within the nested loop.